.github/detect-secrets.yaml

```diff
@@ -0,0 +1,61 @@
+name: "Secret Detection"
+on:
+  push:
+    branches: [main]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [main]
+
+jobs:
+  secret-detection:
+    name: Secret-Detection
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+    - name: Upgrade tooling
+      run: |
+        python3 -m pip install --upgrade pip
+        pip install --upgrade git+https://github.com/Yelp/detect-secrets.git
```

Comment:
Should we be pinning this to a specific version to avoid this test from breaking in the future? Also, any particular reason why we're using the GitHub repo instead of using the PyPi package?

https://pypi.org/project/detect-secrets/

Reply:
Not sure why the SLIM project made the choices they did when they created this file, but I'm fine with making this change.